cmd/cstor-pool-mgmt/controller/pool-controller/handler.go

```diff
@@ -73,140 +73,151 @@ func (c *CStorPoolController) cStorPoolEventHandler(operation common.QueueOperat
-
-			// CreatePool is to create cstor pool.
-			err = pool.CreatePool(cStorPoolGot)
-			if err != nil {
-				glog.Errorf("Pool creation failure: %v", string(cStorPoolGot.GetUID()))
-				c.recorder.Event(cStorPoolGot, corev1.EventTypeWarning, string(common.FailureCreate), string(common.MessageResourceFailCreate))
-				return string(apis.CStorPoolStatusOffline), err
-			}
-			glog.Infof("Pool creation successful: %v", string(cStorPoolGot.GetUID()))
-			c.recorder.Event(cStorPoolGot, corev1.EventTypeNormal, string(common.SuccessCreated), string(common.MessageResourceCreated))
-			return string(apis.CStorPoolStatusOnline), nil
-		}
-		glog.Infof("Not init status: %v, %v", cStorPoolGot.ObjectMeta.Name, string(cStorPoolGot.GetUID()))
-
-		break
-
-	case common.QOpDestroy:
-		glog.Infof("Processing cStorPool Destroy event %v, %v", cStorPoolGot.ObjectMeta.Name, string(cStorPoolGot.GetUID()))
-		// DeletePool is to delete cstor pool.
-		err := pool.DeletePool(string(pool.PoolPrefix) + string(cStorPoolGot.ObjectMeta.UID))
-		if err != nil {
-			c.recorder.Event(cStorPoolGot, corev1.EventTypeWarning, string(common.FailureDestroy), string(common.MessageResourceFailDestroy))
-			return string(apis.CStorPoolStatusDeletionFailed), err
-		}
+	// make a check if initialImportedPoolVol is not empty, then notify cvr controller
+	// through channel.
+	if len(common.InitialImportedPoolVol) != 0 {
+		common.SyncResources.IsImported = true
+	} else {
+		common.SyncResources.IsImported = false
+	}
 
+	// IsInitStatus is to check if initial status of cstorpool object is `init`.
+	if IsInitStatus(cStorPoolGot) {
 		// LabelClear is to clear pool label
 		err = pool.LabelClear(cStorPoolGot.Spec.Disks.DiskList)
 		if err != nil {
 			glog.Errorf(err.Error(), cStorPoolGot.GetUID())
+		} else {
+			glog.Infof("Label clear successful: %v", string(cStorPoolGot.GetUID()))
 		}
-		glog.Infof("Label clear successful: %v", string(cStorPoolGot.GetUID()))
 
-		// removeFinalizer is to remove finalizer of cStorPool resource.
-		err = c.removeFinalizer(cStorPoolGot)
+		// CreatePool is to create cstor pool.
+		err = pool.CreatePool(cStorPoolGot)
 		if err != nil {
+			glog.Errorf("Pool creation failure: %v", string(cStorPoolGot.GetUID()))
+			c.recorder.Event(cStorPoolGot, corev1.EventTypeWarning, string(common.FailureCreate), string(common.MessageResourceFailCreate))
 			return string(apis.CStorPoolStatusOffline), err
 		}
-		return "", nil
+		glog.Infof("Pool creation successful: %v", string(cStorPoolGot.GetUID()))
+		c.recorder.Event(cStorPoolGot, corev1.EventTypeNormal, string(common.SuccessCreated), string(common.MessageResourceCreated))
+		return string(apis.CStorPoolStatusOnline), nil
 	}
+	glog.Infof("Not init status: %v, %v", cStorPoolGot.ObjectMeta.Name, string(cStorPoolGot.GetUID()))
+
+	return string(apis.CStorPoolStatusOffline), importPoolErr
+}
+
+func (c *CStorPoolController) cStorPoolDestroyEventHandler(cStorPoolGot *apis.CStorPool) (string, error) {
```

Comment:
Like the idea of having a separate handler for pool destroy. Need to handle cases like, all the volumes of the pool should be destroyed before calling zpool destroy.

Reply:
will add this as feature request and handle this in new PR. This involves listing all cvr CRs and updating the status